src/Mod/Sketcher/App/SketchObject.cpp

```diff
@@ -7106,15 +7111,92 @@ int SketchObject::carbonCopy(App::DocumentObject* pObj, bool construction)
         solverNeedsUpdate = true;
     }
 
+    auto applyGeometryFlipCorrection = [xinv, yinv, origin, axisV, axisH]
+                                       (Part::Geometry* geoNew) {
+        if (!(xinv || yinv)) {

```

Comment:
Let's not complicate logic too much - nested logic is harder to process.

```suggestion
        if (!xinv && !yinv) {
```